catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/ScriptFragment.kt

```diff
@@ -0,0 +1,1009 @@
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        menu.findItem(R.id.show_details).isVisible = false
+        menu.findItem(R.id.rename).isVisible = false
+        menu.findItem(R.id.catblocks_reorder_scripts).isVisible = false
+        menu.findItem(R.id.find).isVisible = true
+        if (!BuildConfig.FEATURE_CATBLOCKS_ENABLED) {
+            menu.findItem(R.id.catblocks).isVisible = false
+        }
+        super.onPrepareOptionsMenu(menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        if (listView?.isCurrentlyMoving == true) {
+            listView?.highlightMovingItem()
+            return true
+        }
+        if (listView?.isCurrentlyHighlighted == true) {
+            listView?.cancelHighlighting()
+        }
+        when (item.itemId) {
+            R.id.menu_undo -> loadProjectAfterUndoOption()
+            R.id.backpack -> prepareActionMode(BACKPACK)
+            R.id.copy -> prepareActionMode(COPY)
+            R.id.delete -> prepareActionMode(DELETE)
+            R.id.comment_in_out -> prepareActionMode(COMMENT)
+            R.id.catblocks -> switchToCatblocks()
+            R.id.find -> scriptFinder?.open()
+            else -> return super.onOptionsItemSelected(item)
+        }
+        return true
+    }
+
+    fun notifyDataSetChanged() {
+        adapter.notifyDataSetChanged()
+    }
+
+    val isCurrentlyMoving: Boolean
+        get() = listView?.isCurrentlyMoving ?: false
+
+    fun highlightMovingItem() {
+        listView?.highlightMovingItem()
+    }
+
+    fun cancelMove() {
+        listView?.cancelMove()
+        val sprite = projectManager.currentSprite
+        adapter.updateItems(sprite)
+    }
+
+    val isCurrentlyHighlighted: Boolean
+        get() = listView?.isCurrentlyHighlighted ?: false
+
+    fun cancelHighlighting() {
+        listView?.cancelHighlighting()
+    }
+
+    private fun showCategoryFragment() {
+        val brickCategoryFragment = BrickCategoryFragment()
+        brickCategoryFragment.setOnCategorySelectedListener(this)
+
+        fragmentManager?.beginTransaction()?.add(
+            R.id.fragment_container,
+            brickCategoryFragment,
+            BrickCategoryFragment.BRICK_CATEGORY_FRAGMENT_TAG
+        )?.addToBackStack(BrickCategoryFragment.BRICK_CATEGORY_FRAGMENT_TAG)?.commit()
+        SnackbarUtil.showHintSnackbar(activity, R.string.hint_category)
+    }
+
+    override fun onCategorySelected(category: String?) {
+        val addListFragment: ListFragment
+        val tag: String
+        if (category == context?.getString(R.string.category_user_bricks)) {
+            addListFragment = UserDefinedBrickListFragment.newInstance(this)
+            tag = UserDefinedBrickListFragment.USER_DEFINED_BRICK_LIST_FRAGMENT_TAG
+        } else {
+            addListFragment = AddBrickFragment.newInstance(category, this)
+            tag = AddBrickFragment.ADD_BRICK_FRAGMENT_TAG
+        }
+
+        fragmentManager?.beginTransaction()
```

Comment:
fragmentManager is Deprecated, use parentFragmentManager instead. then remove Unnecessary safe call '?'
```suggestion
        parentFragmentManager.beginTransaction()
```